src/MessageBox/MessageBox.js

```diff
@@ -0,0 +1,176 @@
+ * UI, such as technical errors. This component inherits the look and behavior
+ * of the **Dialog** (excluding dragging/resizing).
+ *
+ * @returns {Node} MessageBox component
+ */
+function MessageBox({
+    actions,
+    backdropClassName,
+    bodyProps,
+    children,
+    className,
+    contentProps,
+    cssNamespace,
+    footerProps,
+    headerProps,
+    onClose,
+    show,
+    size,
+    title,
+    type,
+    ...props
+}) {
+    const bodyElm = document.querySelector('body');
+
+    const handleKeyPress = event => {
+        if (event.key === 'Escape' || event.key === 'Esc') {
+            onClose(event);
+        }
+    };
+
+    useEffect(() => {
+        document.addEventListener('keydown', handleKeyPress, false);
+        return () => document.removeEventListener('keydown', handleKeyPress, false);
+    }, []);
+
+    let icon;
+    if (type === 'confirmation') {
+        icon = 'question-mark';
+    } else if (type === 'error') {
+        icon = 'message-error';
+    } else if (type === 'success') {
+        icon = 'message-success';
+    } else if (type === 'warning') {
+        icon = 'message-warning';
+    } else if (type === 'information') {
+        icon = 'message-information';
+    }
+
+    if (!show) {
+        return null;
+    }
+
+    const boxClasses = classnames(
+        `${cssNamespace}-message-box`,
+        {
+            [`${cssNamespace}-message-box--${type}`]: type !== 'default',
+            [`${cssNamespace}-message-box--active`]: show
+        },
+        className,
+    );
+
+    const contentClasses = classnames(
+        `${cssNamespace}-message-box__content`,
+        {
+            [`${cssNamespace}-message-box__content--${size}`]: size
+        },
+    );
+
+    const bodyClasses = classnames(
+        `${cssNamespace}-message-box__body`
+    );
+
+    return ReactDOM.createPortal(
+        <FocusLock
+            as='div'
+            className={backdropClassName}
+            lockProps={{ ...props }}>
+            <span data-autofocus tabIndex='-1' />
+            <div className={boxClasses}>
+                <section className={contentClasses} {...contentProps}>
+                    <Bar
+                        className={classnames(`${cssNamespace}-message-box__header`)}
+                        leftComponents={[<>
+                            {icon && <Icon glyph={icon} />}
+                            <Title level={2} levelStyle={5}>{title}</Title>
+                        </>]}
+                        type='header'
+                        { ...headerProps } />
+                    <section className={bodyClasses} {...bodyProps}>
+                        {children}
+                    </section>
+                    <Bar
+                        className={classnames(`${cssNamespace}-message-box__footer`)}
+                        rightComponents={actions.map(child => cloneElement(child, {
+                            className: classnames(`${cssNamespace}-message-box__decisive-button`),
```

Comment:
we should use the className coming from the child and this one should be only used as a fallback.

Reply:
i got inspired by an identical solution used in the `Dialog` component. Apparently this class is expected to be there. I could perhaps add user-provided classes so this one gets appended, rather than replaced.